src/ImageSharp/Formats/Gif/LzwDecoder.cs

```diff
@@ -232,6 +232,159 @@ public void DecodePixels(int minCodeSize, Buffer2D<byte> pixels)
         }
     }
 
+    /// <summary>
+    /// Decodes and decompresses all pixel indices from the stream allowing skipping of the data.
+    /// </summary>
+    /// <param name="minCodeSize">Minimum code size of the data.</param>
+    /// <param name="length">The resulting index table length.</param>
+    public void SkipIndices(int minCodeSize, int length)
```

Comment:
Gif doesn't tell you in advance the length of the combined LZW segment blocks. You have to read each one which must be decoded to read the next one.